torch/csrc/jit/passes/quantization/helper.cpp

```diff
@@ -309,10 +309,16 @@ std::vector<Value*> getPassThroughInputs(Value* v) {
       inputs.push_back(v);
     }
     return inputs;
-  } else if (n->kind() == Symbol::aten("append")) {
-    // notice that append is an op that changes input inplace
-    return {n->input(0), n->input(1)};
+  } else if (isListAdd(n)) {
```

Comment:
What is the rule for list add? Do we assume that all the tensors in the list have a dequantize op prior to the add?

Reply:
this is the transformed "append", as shown in the description. we will check if the inputs are produced with dequantize to make sure the inputs are quantized

Reply:
the rule of list add:
```
%y = aten::add(%list, %x)
```
we'll check if %list is empty list, if it is, then the pass through list for %y is {%x}, otherwise, it is {%list, %x}

Reply:
How do you check if the existing list has only dequantized tensors?. I see how you can do it for the input %x, dont follow how the check is done for %list

Reply:
There are two cases for %list:
1. 
one case is same as %x
```
%list = aten::dequantize(%list_quant)
%result = aten::add(%list, %x_list)
```
we check if the list is produced by dequantize or not
2.
another case is when list is empty, it can be considered as containing quantized tensors
```
%list : Tensor[] = prim::ListConstruct()
%result = aten::add(%list, %x_list)
```
in this case we only need to check if %x_list is produced by dequantize.

Reply:
My question is on how we check if %list is produced by dequantize or not. Do we iterate over all elements in the list?

Reply:
no we don't, we just check if %list is produced by dequantize or not